### PR TITLE
Improve Markdown files

### PR DESCRIPTION
**Pull request by ghost**
Done so far:
- wrap all `.md` files at 80 characters
- fix Markdown errors
- fix typos (not thoroughly)

Todo:
- [ x ] standardise Markdown syntax (partially depends on the next point being accepted or not) -> will be a different PR.

Proposed:
- continue formatting according to this style guide http://www.cirosantilli.com/markdown-style-guide

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,61 +1,78 @@
 # Filing an issue
 
 ### Must read
-* If you aren't sure, you can ask on the [**IRC channel**](https://webchat.freenode.net/?channels=qtox) or read our [**wiki**](https://github.com/tux3/qTox/wiki) first.
+* If you aren't sure, you can ask on the
+  [**IRC channel**](https://webchat.freenode.net/?channels=qtox) or read our
```
Comment (zetok) [inline CONTRIBUTING.md:5]: `[word](link)` can be actually split across lines, i.e.

```
[word]
(link)
```

Comment (ghost) [inline CONTRIBUTING.md:5]: I think that the raw file is easier to understand when you leave it together.

Also, in GH's online editor and Atom, separating it breaks the code highlighting:

without line break

with line break

Comment (zetok) [inline CONTRIBUTING.md:5]: > Also, in GH's online editor and Atom, separating it breaks the code highlighting:

Same goes for vim. Who cares, as long as it actually works ;)

Comment (ghost) [inline CONTRIBUTING.md:5]: I tried the change in `CONTRIBUTING` and `INSTALL`, it brings little gain. In most cases, either the URL alone is longer than the remaining columns, or the final amount of lines is the same breaking the `[]()` construct than leaving it together. I'd leave the remaining cases untouched for consistency.

**Comment by ghost** [comment]
> http://www.cirosantilli.com/markdown-style-guide

What do you thing about that?

**Comment by sudden6** [comment]
This is a huge document, which options did you intend to use for our MD files?

File: CONTRIBUTING.md (hunk 1 above)
Comment (ghost) [inline CONTRIBUTING.md:5]: Google doesn't break links either: https://github.com/google/styleguide/blob/gh-pages/docguide/style.md#character-line-limit

**Comment by ghost** [comment]
It has a lot of blah blah, examples and variants. The actual changes won't be that many, because most are just common sense. A ~exhaustive summary of my  proposed changes/standards to apply from the guide:
- Don’t use 2 or more consecutive empty lines [...] except where they must appear literally
- Don’t prefix shell code with dollar signs `$` unless you will be showing the command output on the same code block.
- What to mark as code: executables, file paths, version numbers, capitalized explanation of abbreviations, terms related to computers that you don’t want to add to your dictionary
- Surround headers by a single empty line except at the beginning of the file.
- Don’t add a trailing colon `:` / trailing period `.` to headers.
- Indentation of content inside lists.
- If every item of a list is a single line long, don’t add empty lines between items. Otherwise, add empty lines between every item.
- Surround lists by one empty line.
- Each list item has the same case as it would have if it were concatenated with the sentence that comes before the list.
  Punctuate at the end of list items if either it contains multiple sentences or paragraphs or starts with an upper case letter.
- Tables (all recommendations)
- Separate consecutive elements with an empty HTML comment `<!-- -->`.
- Bold and italic with asterisk format.
- Don’t use automatic links without angle brackets.

space-sentence:1
header:setex
wrap:space
list-marker:asterisk
list-space:1
code:fenced

**Comment by zetok** [comment]
> Don’t use 2 or more consecutive empty lines [...] except where they must appear literally

I'm not going to adhere to this one. I.e. I am going to keep using 2 empty lines when I see a need for them.

> Surround headers by a single empty line except at the beginning of the file.

I usually try to put 2 empty lines before header. And unless there's a good reason not to, I'm going to stick to it.

# 

Aside from that, why not.

**Comment by sudden6** [comment]
> > Don’t use 2 or more consecutive empty lines [...] except where they must appear literally
> 
> I'm not going to adhere to this one. I.e. I am going to keep using 2 empty lines when I see a need for them.

@zetok I think header:setex makes two empty lines really unnecessary in the most cases

> Separate consecutive elements with an empty HTML comment .

@kehugter can you explain the purpose of that?

**Comment by ghost** [comment]
@sudden6 improves readability in this kind of scenario:

```
* list 1 el 1
* list 1 el 2
* list 1 el 3
  * list 1 el 3 sub 1
  * list 1 el 3 sub 2

* list 2 el 1
* list 2 el 2
  * list 2 el 2 sub 1
* list 2 el 3
```

vs

```
* list 1 el 1
* list 1 el 2
* list 1 el 3
  * list 1 el 3 sub 1
  * list 1 el 3 sub 2

<!-- -->

* list 2 el 1
* list 2 el 2
  * list 2 el 2 sub 1
* list 2 el 3
```

**Comment by sudden6** [comment]
ok, then I also support this.

**Comment by zetok** [comment]
Well, inserting `<!-- -->` might make things clearer, but at the same time it could make contributing harder.

Instead of it, a few empty lines can always be inserted, which will have ~same effect on visibility, and won't cause any problems.

E.g. someone not familiar with HTML/markdown would have to think what `<!-- -->` means:

```
* list 1 el 1
* list 1 el 2
* list 1 el 3
  * list 1 el 3 sub 1
  * list 1 el 3 sub 2

<!-- -->

* list 2 el 1
* list 2 el 2
  * list 2 el 2 sub 1
* list 2 el 3
```

On the other hand, using just empty lines is quite clear to anyone, regardless of their knowledge about HTML/markdown:

```
* list 1 el 1
* list 1 el 2
* list 1 el 3
  * list 1 el 3 sub 1
  * list 1 el 3 sub 2


* list 2 el 1
* list 2 el 2
  * list 2 el 2 sub 1
* list 2 el 3
```

Thus, I'm against _having to_ use it `<!-- -->`.

# 

Aside from that, @kehugter, how ready is the PR for a merge? I.e. as time flows, `master` will be receiving more changes that will require this PR to be rebased – there are also a few changes that I'd like to make to .md files that aren't that important, but would cause a few conflicts with this PR, so I'm kinda waiting for it. Unless you don't mind rebasing & fixing your PR every time something breaking lands on master.. IMO though it would be better to merge PR as soon as it's good for merge, and eventual discussion about what formatting should be more to a dedicated issue for it.

Note that for me `good for merge` can consist only of formatting change, with little to no changes to the content – any fixes to the content can always be made as a new PR :)

**Comment by sudden6** [comment]
Ok, seeing @zetok s arguments, I like this variant more than `<!-- -->`

**Comment by ghost** [comment]
@zetok yes, this can be merged. Nothing was changed in the logic, only the form.

**Pull request merged**